File: email-front/src/app/email/email.ts
export class Email {
    private from: string;
    private to: Array<string>;
    private subject: string;
    private body: string;
    private date: Date;
    private read: boolean;
    private priority: number;
    private attachment: FormData;

    constructor(from: string, to: Array<string>, subject: string, body: string, read: boolean, priority: number, attachment: FormData){
        this.from = from;
        this.to = to;
        this.subject = subject;
        this.body = body;
        this.read = read;
        this.priority = priority;
        this.attachment = attachment;
        this.date = new Date;
    }

    public static createEmailFromBack(obj: any){
        return new Email(obj.from, obj.to, obj.subject, obj.body, obj.read, obj.priority, obj.attachment);
    }

    getFrom(){
        return this.from;
    }
    getTo(){
        return this.to;
    }
    getSubject(){
        return this.subject;
    }
    getBody(){
        return this.body;
    }
    getDate(){
        return this.date.getDate() + '/' + (this.date.getMonth() + 1 ) + '/' + this.date.getFullYear();
    }
    getPriority(){
        return this.priority;
    }
    getRead(){
        return this.read;
    }
    getAttachment(){
        return this.attachment;
    }
    toggleRead(){
        this.read = !this.read;
    }
}

File: email/email/src/main/java/com/CSED/email/Email/Email.java
package com.CSED.email.Email;

import java.util.ArrayList;
import java.util.Date;
import java.util.LinkedList;
import java.util.Queue;

public class Email implements Comparable<Email> {
    private String from;
    private Queue<String> to = new LinkedList<>();
    private String subject;
    private String body;
    private String[] attachment;
    private Integer priority;
    private boolean read;
    private Date date;

    public Email(EmailBuilder builder){
        this.from = builder.getSender();
        this.to.addAll(builder.getReceiver());
        this.subject = builder.getSubject();
        this.body = builder.getBody();
        this.attachment = builder.getAttachment();
        this.priority = Integer.valueOf(builder.getPriority());
        this.read = builder.isRead();
        this.date = builder.getDate();
    }
    public Email(){

    }
    @Override
    public int compareTo(Email email){
        return this.getPriority().compareTo(email.getPriority());
    }

    public String getFrom() {
        return from;
    }

    public void setFrom(String sender) {
        this.from = sender;
    }

    public Queue<String> getTo() {
        return to;
    }

    public void setTo(Queue<String> receiver) {
        this.to = receiver;
    }

    public String getSubject() {
        return subject;
    }

    public void setSubject(String subject) {
        this.subject = subject;
    }

    public String getBody() {
        return body;
    }

    public void setBody(String body) {
        this.body = body;
    }

    public String[] getAttachment() {
        return attachment;
    }

    public void setAttachment(String[] attachment) {
        this.attachment = attachment;
    }
    public Integer getPriority() {
        return priority;
    }

    public void setPriority(Integer priority) {
        this.priority = priority;
    }

    public boolean isRead() {
        return read;
    }

    public void setRead(boolean read) {
        this.read = read;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

}